src/persist-client/src/cli/inspect.rs

```diff
@@ -706,33 +717,41 @@ pub async fn arrow_rs_roundtrip(blob_uri: &str) -> Result<impl Serialize, anyhow
 
     let mut report = BlobReport::default();
     for (shard_id, partial_keys) in shards {
-        // Drives all of the GET futures concurrently.
+        // Using a bounded channel prevents us from fetching too many blobs at once.
```

Comment:
Maybe https://docs.rs/tokio/latest/tokio/sync/struct.Semaphore.html?

Reply:
FWIW the restore-blob command uses `flat_map_unordered` to solve a similar issue, in case having another option is helpful and not annoying. 😅